lib/Wikia/src/CircuitBreaker/ServiceCircuitBreaker.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Wikia\CircuitBreaker;
+
+class ServiceCircuitBreaker {
+	/** @var string */
+	private $serviceName;
+
+	/** @var CircuitBreakerStorage */
+	private $circuitBreaker;
+
+	/**
+	 * ServiceCircuitBreaker constructor.
+	 * @param $serviceName string name of the service
+	 * @param CircuitBreakerStorage $circuitBreaker
+	 */
+	public function __construct( string $serviceName, CircuitBreakerStorage $circuitBreaker ) {
+		$this->serviceName = $serviceName;
+		$this->circuitBreaker = $circuitBreaker;
+	}
+
+	/**
+	 * @param $status bool status of the operation
+	 * @return bool
+	 */
+	public function SetOperationStatus( $status ) {
+		return $this->circuitBreaker->setOperationStatus( $this->serviceName, $status );
+	}
+
+	/**
+	 * @throws CircuitBreakerOpen
+	 */
+	public function AssertOperationAllowed() {
+		if ( !$this->OperationAllowed() ) {
+			throw new CircuitBreakerOpen( $this->serviceName );
+		}
+	}
+
+	/**
+	 * @return bool returns true if operation can be performed on a given service
+	 */
+	public function OperationAllowed() {
```

Comment:
Coding conventions, apply to all calls:
```suggestion
	public function operationAllowed() {
```